src/Post.php

```diff
@@ -2,40 +2,23 @@
 
 use Leean\Endpoints\Inc\Content;
 use Leean\Endpoints\Inc\Type;
+use Leean\AbstractEndpoint;
 
 /**
  * Class to provide activation point for our endpoints.
  */
-class Post
-{
-	const ENDPOINT = '/post';
+class Post extends AbstractEndpoint {
 
 	/**
-	 * Init.
+	 * Slug for the definition of the post.
+	 *
+	 * @Override
+	 * @var String
 	 */
-	public static function init() {
-		add_action( 'rest_api_init', function () {
-			$namespace = apply_filters( 'ln_endpoints_api_namespace', 'leean', self::ENDPOINT );
-			$version = apply_filters( 'ln_endpoints_api_version', 'v1', self::ENDPOINT );
+	protected $endpoint = '/post';
 
-			register_rest_route(
-				$namespace . '/' . $version,
-				self::ENDPOINT,
-				[
-					'methods' => 'GET',
-					'callback' => [ __CLASS__, 'get_post' ],
-					'args' => [
-						'slug' => [
-							'required' => true,
-							'sanitize_callback' => function ( $param, $request, $key ) {
-								return sanitize_text_field( $param );
-							},
-						],
-					],
-				]
-			);
-		} );
-	}
+	const QUERY_FILTER = 'ln_endpoints_query_args';
```

Comment:
Agreed it makes more sense. I’m going to update as required. 

Thanks

> On Mar 23, 2016, at 8:10 AM, Adam Fenton <notifications@github.com> wrote:
> 
> In src/Post.php <https://github.com/moxie-leean/wp-endpoints-post/pull/8#discussion_r57164677>:
> 
> > -				[
> > -					'methods' => 'GET',
> > -					'callback' => [ __CLASS__, 'get_post' ],
> > -					'args' => [
> > -						'slug' => [
> > -							'required' => true,
> > -							'sanitize_callback' => function ( $param, $request, $key ) {
> > -								return sanitize_text_field( $param );
> > -							},
> > -						],
> > -					],
> > -				]
> > -			);
> > -		} );
> > -	}
> > +	const QUERY_FILTER = 'ln_endpoints_query_args';
> Shouldn't this be ln_endpoints_post_query_args?
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub <https://github.com/moxie-leean/wp-endpoints-post/pull/8/files/ec67f3f62b2df980060d2facaefa13ffe710debf#r57164677>